packages/shared/src/telemetry/types.ts

```diff
@@ -71,5 +71,6 @@ export type TelemetryEvent = {
 export type TelemetryEventRaw<Payload = TelemetryEvent['payload']> = {
   event: TelemetryEvent['event'];
   eventSamplingRate?: number;
+  clientCacheKey?: `${TelemetryEvent['event']}:${string}`;
```

Comment:
I'm not 100% decided if it's better to apply it as a required property. Depending on its nature, there's the likelihood of an event being called on the client side and server side.

`clerkMiddleware` gets only called on the server side, using an event-specific sampling rate. However, a React hook could be called on a client component.

Reply:
Alright, changing the approach here - we're going to generate that cache key based on the event payload, instead of having to pass it as a property.